src/tests/livepatch.c

```diff
@@ -0,0 +1,176 @@
+		goto on_error_put;
+	}
+
+	if (!c->createl(c, "busybox", NULL, NULL, 0, NULL)) {
+		lxc_error("%s\n", "Failed to create busybox container \"livepatch\"");
+		goto on_error_put;
+	}
+
+	if (!c->is_defined(c)) {
+		lxc_error("%s\n", "Container \"livepatch\" is not defined");
+		goto on_error_put;
+	}
+
+	c->clear_config(c);
+
+	if (!c->load_config(c, NULL)) {
+		lxc_error("%s\n", "Failed to load config for container \"livepatch\"");
+		goto on_error_destroy;
+	}
+
+	if (!c->want_daemonize(c, true)) {
+		lxc_error("%s\n", "Failed to mark container \"livepatch\" daemonized");
+		goto on_error_destroy;
+	}
+
+	if (!c->startl(c, 0, NULL)) {
+		lxc_error("%s\n", "Failed to start container \"livepatch\" daemonized");
+		goto on_error_destroy;
+	}
+
+	/* Test whether the current value is ok. */
+	value = c->get_running_config_item(c, "lxc.net.0.type");
+	if (!value) {
+		lxc_error("%s\n", "Failed to retrieve running config item \"lxc.net.0.type\"");
+		goto on_error_destroy;
+	}
+
+	if (strcmp(value, "veth")) {
+		lxc_error("Retrieved unexpected value for config item "
+			  "\"lxc.net.0.type\": veth != %s", value);
+		free(value);
+		goto on_error_destroy;
+	}
+	free(value);
+
+	/* Change current in-memory value. */
+	if (!c->set_running_config_item(c, "lxc.net.0.type", "macvlan")) {
+		lxc_error("%s\n", "Failed to set running config item "
+				  "\"lxc.net.0.type\" to \"macvlan\"");
+		goto on_error_destroy;
+	}
+
+	/* Verify change. */
+	value = c->get_running_config_item(c, "lxc.net.0.type");
+	if (!value) {
+		lxc_error("%s\n", "Failed to retrieve running config item \"lxc.net.0.type\"");
+		goto on_error_destroy;
+	}
+
+	if (strcmp(value, "macvlan")) {
+		lxc_error("Retrieved unexpected value for config item "
+			  "\"lxc.net.0.type\": macvlan != %s", value);
+		free(value);
+		goto on_error_destroy;
+	}
+	free(value);
+
+	/* Change current in-memory value. */
+	if (!c->set_running_config_item(c, "lxc.net.0.type", "veth")) {
+		lxc_error("%s\n", "Failed to set running config item "
+				  "\"lxc.net.0.type\" to \"veth\"");
+		goto on_error_destroy;
+	}
+
+	/* Add new in-memory value. */
+	if (!c->set_running_config_item(c, "lxc.net.1.type", "veth")) {
+		lxc_error("%s\n", "Failed to set running config item "
+				  "\"lxc.net.1.type\" to \"veth\"");
+		goto on_error_destroy;
+	}
+
+	/* Verify change. */
+	value = c->get_running_config_item(c, "lxc.net.1.type");
+	if (!value) {
+		lxc_error("%s\n", "Failed to retrieve running config item \"lxc.net.1.type\"");
+		goto on_error_destroy;
+	}
+
+	if (strcmp(value, "veth")) {
+		lxc_error("Retrieved unexpected value for config item "
+			  "\"lxc.net.1.type\": veth != %s", value);
+		free(value);
+		goto on_error_destroy;
+	}
+	free(value);
+
+	/* Remove in-memory value. */
+	if (!c->set_running_config_item(c, "lxc.net.1.type", "")) {
+		lxc_error("%s\n", "Failed to clear running config item "
+				  "\"lxc.net.1.type\"");
+		goto on_error_destroy;
```

Comment:
Can you add a test having the container reboot, and make sure it comes up right?

And maybe also, separately, stop+start it?

Reply:
Yeah, can do.